services/order.py

```diff
@@ -0,0 +1,60 @@
+from typing import Any
+from datetime import datetime
+
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, User, MovieSession, Ticket
+
+
+def get_orders(
+        username: str = None
+) -> QuerySet:
+    queryset = Order.objects.all().order_by("-id")
+    if username:
+        queryset = queryset.filter(user__username=username).order_by("-id")
+
+    return queryset
+
+
+def create_order(
+    tickets: list[dict[str, Any]],
+    username: str,
+    date: str = None
+) -> Order:
+    try:
+        user = User.objects.get(username=username)  # Retrieve the user
+    except User.DoesNotExist:
+        raise ValidationError(f"User with username"
+                              f" '{username}' does not exist.")
+
+    with transaction.atomic():
+        # Create the order
+        order = Order.objects.create(user=user)
+        if date:
+            created_at = datetime.strptime(date, "%Y-%m-%d %H:%M")
```

Comment:
Consider validating the `date` string format before parsing it with `datetime.strptime` to ensure it matches the expected format and avoid potential errors.